control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2929,6 +2939,31 @@ func deleteIfExists(ctx context.Context, c client.Client, o client.Object) error
 	return nil
 }
 
+// removeServiceCAAnnotationAndSecret will delete Secret 'secret' and
+// remove the annotation "service.beta.openshift.io/serving-cert-secret-name"
+// from Service 'service' if it contains this annotation.
+// This is used to remove Secrets generated by the service-ca in case
+// of upgrade, from a control-plane version using service-ca generated certs
+// to a version where the service uses HCP controller generated certs.
+func removeServiceCAAnnotationAndSecret(ctx context.Context, c client.Client, service *corev1.Service, secret *corev1.Secret) error {
+	if err := c.Get(ctx, client.ObjectKeyFromObject(service), service); err != nil {
+		return fmt.Errorf("failed to get service: %w", err)
+	}
+	_, ok := service.Annotations["service.beta.openshift.io/serving-cert-secret-name"]
+	if ok {
+		delete(service.Annotations, "service.beta.openshift.io/serving-cert-secret-name")
+		err := c.Update(ctx, service)
+		if err != nil {
+			return fmt.Errorf("failed to update service: %w", err)
+		}
+		err = deleteIfExists(ctx, c, secret)
```

Comment:
Hmmm, I see your point. The problem is that this is the same secret name that is now going to be generated by the control-plane-operator PKI package, so if we always delete it here then it will be deleted/recreated on every reconciliation.

Maybe we can instead check for a service-ca specific annotation on this secret before deleting it (rather than only deleting it if the *service* has an annotation. Something like
`service.beta.openshift.io/originating-service-name` or `service.beta.openshift.io/expiry`?

Reply:
@dagrayvid that makes sense to me. The old secret should always have a `service.beta.openshift.io/originating-service-name` annotation but the new one shouldn't

Reply:
Okay, I added this. Looks like e2e failed for an unrelated reason.